cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -142,17 +142,19 @@ public String toString() {
 
   /**
    * Abstract base class for composite trigger in REST requests/responses.
+   *
+   * @param <T> type of triggers contained in the composite trigger
    */
-  public abstract static class AbstractCompositeTrigger extends ProtoTrigger {
-    private final List<Trigger> triggers;
+  public abstract static class AbstractCompositeTrigger<T extends Trigger> extends ProtoTrigger {
+    private final List<T> triggers;
```

Comment:
@albertshau I just realized that this trigger is returned as JSON object for GET schedules request, so this will not be backward compatible. Should I keep `ProtoTrigger. AndTrigger` and `ProtoTrigger. OrTrigger` to accept `Trigger` and convert the schedules to contain ProtoTrigger in the response?